packages/agents/lighthouse/src/tasks/processFromRoot/helpers/xlayer.ts

```diff
@@ -0,0 +1,210 @@
+  logger.info("Method start", requestContext, methodContext);
+
+  const xlayerBridgeApiEndpoint =
+    hubChainId === 1 ? "https://rpc.xlayer.tech/priapi/v1/ob/bridge/" : "https://testrpc.x1.tech/priapi/v1/ob/bridge";
+
+  const spokeConnector = contracts.spokeConnector(
+    spokeChainId ?? 0,
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+
+  const hubConnector = contracts.hubConnector(
+    hubChainId ?? 0,
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+
+  const spokeDeposits = await getDeposits(xlayerBridgeApiEndpoint, spokeConnector!.address);
+  const hubDeposits = await getDeposits(xlayerBridgeApiEndpoint, hubConnector!.address);
+  const claimableDeposits = spokeDeposits
+    .concat(hubDeposits)
+    .filter((d: any) => d.ready_for_claim && d.claim_tx_hash === "");
+
+  if (!claimableDeposits.length) {
+    throw new NoRootAvailable(spokeChainId, hubChainId, requestContext, methodContext, {
+      error: `Claimable desposits don't exist!`,
+    });
+  }
+
+  const hubRpcProvider = new ethers.providers.JsonRpcProvider(hubProvider);
+  const hubConnectorContract = getContract(hubConnector!.address, hubConnector!.abi as any[], hubRpcProvider);
+  const hubAMBAddress: string = await hubConnectorContract.AMB();
+  const spokeRpcProvider = new ethers.providers.JsonRpcProvider(spokeProvider);
+  const spokeConnectorContract = getContract(spokeConnector!.address, spokeConnector!.abi as any[], spokeRpcProvider);
+  const spokeAMBAddress: string = await spokeConnectorContract.AMB();
+
+  for (const deposit of claimableDeposits) {
+    const proof = await getMerkleProof(xlayerBridgeApiEndpoint, deposit.deposit_cnt, deposit.network_id);
+    if (!proof) {
+      continue;
+    }
+
+    const args = [
+      proof.merkle_proof,
+      proof.rollup_merkle_proof,
+      deposit.global_index,
+      proof.main_exit_root,
+      proof.rollup_exit_root,
+      deposit.orig_net,
+      deposit.orig_addr,
+      deposit.dest_net,
+      deposit.dest_addr,
+      deposit.amount,
+      deposit.metadata,
+    ];
+    const encodedData = new ethers.utils.Interface(PolygonZkEVMBridgeV2ABI).encodeFunctionData("claimMessage", args);
+
+    logger.info("Sending process message from root tx", requestContext, methodContext, {
+      args,
+      encodedData,
+      spokeChain: spokeChainId,
+      hubChain: hubChainId,
+    });
+
+    const isClaimOnL2 = deposit.orig_net === 0;
+    const { taskId } = await sendWithRelayerWithBackup(
+      isClaimOnL2 ? spokeChainId : hubChainId,
+      isClaimOnL2 ? spokeDomainId : hubDomainId,
+      isClaimOnL2 ? spokeAMBAddress : hubAMBAddress,
+      encodedData,
+      relayers,
+      chainreader,
+      logger,
+      requestContext,
+    );
+
+    logger.info("Sent X Layer claim tx to relayer", requestContext, methodContext, { taskId });
+  }
+
+  return [];
```

Comment:
its pretty antipattern to send the relayer task within the `getArguments` function.

is there a better way to structure this? i.e. can the `getProcessArgs` be `getProcessTx` and it includes the `WriteTransaction` struct to send?

Reply:
it seems the only thing it assumes right now that requires this shift is that its sending on the hub. I think if we expand to have `to, domain, data, from, value` then this doesn't need to be unintuitively special-cased